### PR TITLE
Add subscription items

### PR DESCRIPTION
**Pull request by kyleconroy**

### CODE REVIEW

File: subitem/client.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+		params.AppendTo(body)
+
+		lp = &params.ListParams
+		p = params.ToParams()
+	}
+
+	return &Iter{stripe.GetIter(lp, body, func(b *stripe.RequestValues) ([]interface{}, stripe.ListMeta, error) {
+		list := &stripe.SubList{}
+		err := c.B.Call("GET", "/subscriptions", c.Key, b, p, list)
```
Comment (brandur) [inline subitem/client.go]: @kyleconroy I'm not sure whether this is finished, but this doesn't seems quite right. I believe the new endpoint is at `/subscription_items`, and if so, it means that not only is this wrong, but your test suite is either broken or not exhaustive enough to be able to test for the failure.

File: subitem/client.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	var commonParams *stripe.Params
+
+	if params != nil {
+		body = &stripe.RequestValues{}
+		params.AppendTo(body)
+		commonParams = &params.Params
+	}
+
+	item := &stripe.Sub{}
+	err := c.B.Call("DELETE", fmt.Sprintf("/subscriptions_item/%v", id), c.Key, body, commonParams, item)
```
Comment (brandur) [inline subitem/client.go]: And same with here (and the other CRUD endpoints). The patch merged to Ruby was for `subscription_items` (i.e. non-plural subscription), so one of these must be wrong.

File: subitem/client.go (hunk 1 above)
Comment (kyleconroy) [inline subitem/client.go]: Yeah, this PR isn't ready yet. I'll ping you for review once it's ready.

**Comment by kyleconroy** [comment]
@brandur this PR is now ready for review

File: subitem/client_test.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		if err != nil {
+			t.Errorf("del err: %s", err)
+		}
+		if !item.Deleted {
+			t.Errorf("item should be deleted %#v", item)
+		}
+	}
+}
+
+func TestCreateAndDelItem(t *testing.T) {
```
Comment (brandur) [inline subitem/client_test.go]: Was leaving this test empty intentional?

Comment (kyleconroy) [inline subitem/client_test.go]: Whoops, nope. I'll go ahead and fill it out.

**Comment by kyleconroy** [comment]
@brandur added that missing test. Mind taking another look?

**Comment by brandur** [comment]
LGTM.

**Pull request merged**